File: public/dsa/two_pointers/unique_elements_in_sorted_array/optimal/unique_elements_in_sorted_array.cpp

#include <bits/stdc++.h>
using namespace std;

int removeDuplicates(vector<int> &A) {
    int cur = A[0];
    int unique = 1;
    for (int i = 1; i < A.size(); i++) {
        if (A[i] != cur) {
            cur = A[i];
            unique++;
        }
    }
    return unique;
}

// tests-start
int main() {
    vector<int> A = {1, 2, 3, 3, 3, 4, 5, 5};
    int size = removeDuplicates(A);
    cout << size << endl;
    return 0;
}
// tests-end

/*output
5
*/

File: public/dsa/two_pointers/unique_elements_in_sorted_array/optimal/unique_elements_in_sorted_array.py
from typing import List


class Solution:
    def removeDuplicates(self, A: List[int]) -> int:
        cur = A[0]
        unique = 1
        for i in range(1, len(A)):
            if A[i] != cur:
                cur = A[i]
                unique += 1
        return unique


# tests-start
A = [1, 2, 3, 3, 3, 4, 5, 5]
sol = Solution()
print(sol.removeDuplicates(A))
# tests-end

'''output
5
'''
